src/conn/conn_api.c

```diff
@@ -2626,14 +2626,35 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
 
 	WT_ERR(__wt_config_gets(session, cfg, "file_extend", &cval));
 	for (ft = file_types; ft->name != NULL; ft++) {
+		/*
+		 * An unset log extend length is an indicator to set it
+		 * with configured maximum log file size in case of the default
+		 * configuration. Update the log extend length with actual
+		 * desired size when maximum log file size gets initialized as
+		 * part log server initialization.
+		 */
+		conn->log_extend_len = WT_CONFIG_UNSET;
```

Comment:
I think this needs to be above the for-loop. Otherwise `file_extend=(log=1M,data=16M)` would overwrite the set `log_extend_len` when it processed the `data=16M` part on the 2nd time through the loop.

Reply:
It's a good point. Though the "file_types" array defined in this file ensures that data file config always gets parsed before the log file config. That way `log_extend_len` would not get overwritten with` file_extend=(log=1M,data=16M)`. However it makes sense to initialise the log_extend_len above the loop. I have made the change.